### PR TITLE
Internal: Replace ReactDOM.render call in Container empty view with the new createRoot().render

### PR DESCRIPTION
**Pull request by udidol**
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Internal: Replace ReactDOM.render call in Container empty view with the new createRoot().render

## Description

* Used WordPress' best practice for React 18 backwards compatibility support: https://developer.wordpress.org/block-editor/reference-guides/packages/packages-element/

### CODE REVIEW

File: assets/dev/js/editor/controls/base-data.js [hunk 1]
```diff
@@ -205,7 +205,9 @@ ControlBaseDataView = ControlBaseView.extend( {
 		try {
 			return parent && this.container.panel.getControlView( parent );
 			// eslint-disable-next-line no-empty
-		} catch ( e ) {}
+		} catch ( e ) {
+			return null;
```
Comment (yotamselementor) [inline assets/dev/js/editor/controls/base-data.js:209]: Notice that you are changing the behavior here.
Why not just this:
```suggestion
			return;
```

File: assets/dev/js/editor/utils/helpers.js [hunk 2]
```diff
@@ -201,6 +201,7 @@ module.exports = {
 				rendered: false,
 			};
 		}
+		return null;
```
Comment (yotamselementor) [inline assets/dev/js/editor/utils/helpers.js:204]: Here too

File: assets/dev/js/editor/controls/base-data.js (hunk 1 above)
Comment (TzviRabinovitch) [reply assets/dev/js/editor/controls/base-data.js:209]: it throws an error
will double check

File: assets/dev/js/editor/utils/helpers.js (hunk 2 above)
Comment (TzviRabinovitch) [reply assets/dev/js/editor/utils/helpers.js:204]: same as above

File: assets/dev/js/editor/elements/views/container/empty-view.js [hunk 3]
```diff
@@ -35,7 +36,11 @@ export default class EmptyView extends Marionette.ItemView {
 			defaultElement = <EmptyComponent container={ container } />;
 		}
 
-		ReactDOM.render( defaultElement, this.el );
+		if ( createRoot ) {
+			createRoot( this.el ).render( defaultElement );
+		} else {
+			render( defaultElement, this.el );
+		}
```
Comment (Nevoss) [inline assets/dev/js/editor/elements/views/container/empty-view.js:43]:
```suggestion
		if ( ReactDOM.createRoot ) {
			ReactDOM.createRoot( this.el ).render( defaultElement );
		} else {
			ReactDOM.render( defaultElement, this.el );
		}
```

File: assets/dev/js/editor/elements/views/container/empty-view.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import { createRoot, render } from '@wordpress/element';
```
Comment (Nevoss) [inline assets/dev/js/editor/elements/views/container/empty-view.js:1]:
```suggestion
```

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/elements/views/container/empty-view.js (hunk 3 above)
Comment (udidol) [reply assets/dev/js/editor/elements/views/container/empty-view.js]: Done

File: assets/dev/js/editor/utils/helpers.js [hunk 5]
```diff
@@ -163,7 +163,6 @@ module.exports = {
 					rendered: false,
 				};
 			}
-			return;
```
Comment (Nevoss) [inline assets/dev/js/editor/utils/helpers.js:166]: Feels like this `return` statement is intended

**Review by Nevoss — CHANGES_REQUESTED** [review]

**Pull request closed without merging**